### PR TITLE
[#127] Added File Tree Functionality to Display Assignment's Tests Folder from Remote

### PR DESCRIPTION
**Pull request by saksham1919**
Authors: @saksham1919 @Banna150149 

Implemented File Tree functionality to display file browser for assignments.

## Description

Added a File Browser partial.
Modified Controller/ Model/ Helper classes for Assignments to enable File Tree fetching from GitHub and displaying it as a partial on manage assignment page.

## How has this been tested?
RSpec and Cucumber testing done.

image

## Screenshots (if appropriate):
image
image

## Checklist:

- [*] Zero `rubocop` violations
- [*] My code receives an "A" when `rubycritic` is run
- [*] >90% test coverage of my changes
- [*] All new and existing tests passed

### CODE REVIEW

**Review by yaoya2618-dot — APPROVED** [review]

**Comment by masarkar-tamu** [comment]
image

Showing as as a 'Cognitive Complexity' flag on my Code-Climate web plugin. Could this possibly be re-factored?

File: app/views/assignments/_file_tree.html.erb [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+          childrenContainer.classList.toggle('hidden');
+          element.classList.toggle('expanded');
+        }
+      }
+
+
+    </script>
+
+
+    <style>
```
Comment (masarkar-tamu) [inline app/views/assignments/_file_tree.html.erb:29]: I think this section could be shifted to a separate `.css` file.

**Comment by saksham1919** [comment]
> image
> 
> Showing as as a 'Cognitive Complexity' flag on my Code-Climate web plugin. Could this possibly be re-factored?

Thanks for pointing that out. I've refactored the class into multiple methods and updated the RSpec tests accordingly.

File: app/views/assignments/_file_tree.html.erb (hunk 1 above)
Comment (saksham1919) [reply app/views/assignments/_file_tree.html.erb]: Yes, thanks for the suggestion! I've moved that section to a separate .css file.

File: features/file_browser.feature [hunk 2]
```diff
@@ -0,0 +1,18 @@
+Feature: File browser 
+  As a CSCE 120 GitHub organization member
+  So that I can see the File Directory of selected assignment
+  I want to be able to see the file browser
+
+  Background: Assignments in the GitHub organization
+    Given the following assignments exist:
+        | assignment_name | repository_name   |
+        | assignment1     | assignment1       |
+        | assignment2     | assignment2       |
```
Comment (masarkar-tamu) [inline features/file_browser.feature:10]: Why does there need to be multiple assignments in the background information?

Comment (saksham1919) [reply features/file_browser.feature]: Removed them.

**Review by masarkar-tamu — APPROVED** [review]

**Pull request merged**